core/dbt/include/global_project/macros/materializations/incremental/incremental.sql

```diff
@@ -21,8 +21,8 @@
       {% set backup_relation = existing_relation.incorporate(path={"identifier": backup_identifier}) %}
       {% do adapter.drop_relation(backup_relation) %}
 
-      {% do adapter.rename_relation(target_relation, backup_relation) %}
-      {% set build_sql = create_table_as(False, target_relation, sql) %}
+      {% set build_sql = create_table_as(False, tmp_relation, sql) %}
+      {% set need_swap = true %}
```

Comment:
I spent a few minutes comparing the logic here with the logic in the [table materialization](https://github.com/fishtown-analytics/dbt/blob/develop/core/dbt/include/global_project/macros/materializations/table/table.sql), and the most significant difference is that we're creating a temporary relation here instead of a full (intermediate) relation. To that end, and for greater consistency, I recommend switching out the logic here around `tmp_relation` to instead be for `intermediate_relation`:
```sql
      {#-- Make sure the backup doesn't exist so we don't encounter issues with the rename below #}
      {% set tmp_identifier = model['name'] + '__dbt_tmp' %}
      {% set backup_identifier = existing_relation.identifier ~ "__dbt_backup" %}
      
      {% set backup_relation = existing_relation.incorporate(path={"identifier": backup_identifier}) %}
      {% set intermediate_relation = existing_relation.incorporate(path={"identifier": tmp_identifier}) %}

      {% do adapter.drop_relation(intermediate_relation) %}
      {% do adapter.drop_relation(backup_relation) %}

      {% set build_sql = create_table_as(False, intermediate_relation, sql) %}
```

We can also cut line 8, since at this point it's just adding confusion. (The conditional branch below recreates `tmp_relation` for its own purposes.)